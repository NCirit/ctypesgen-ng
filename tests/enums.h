enum color {
	RED,
	GREEN,
	BLUE,
};

enum more_color {
	YELLOW = 3,
	CYAN,
	MAGENTA,
	BLACK = 10,
};

int rank_color(enum color color);

enum color best_color(void);

enum {
	READ,
	WRITE,
	APPEND,
};
