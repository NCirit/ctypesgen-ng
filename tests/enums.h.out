### This file automatically created by ctypesgen-ng, the commandline parameters passed to it are given below
### /app/ctypesgen.py --clang-flags "-Wno-error=implicit-int" --ignore-included --library test.so --headers tests/enums.h

import ctypes
import ctypes.util
import os


if not ('LIBRARY_PATH' in os.environ):
    if 'PYTHONPATH' in os.environ:
        os.environ['LIBRARY_PATH'] = os.environ['PYTHONPATH']
    else:
        os.environ['LIBRARY_PATH'] = os.path.dirname(os.path.abspath(__file__))

if os.name == "nt":
    if 'PYTHONPATH' in os.environ:
        os.environ['PATH'] += os.pathsep + os.environ['PYTHONPATH']
    else:
        os.environ['PATH'] += os.pathsep + os.path.dirname(os.path.abspath(__file__))
test_path = ctypes.util.find_library('test')
test = ctypes.CDLL(os.path.abspath(test_path))

enum_color = ctypes.c_uint
enum_constant_RED = 0
enum_constant_GREEN = 1
enum_constant_BLUE = 2

enum_more_color = ctypes.c_uint
enum_constant_YELLOW = 3
enum_constant_CYAN = 4
enum_constant_MAGENTA = 5
enum_constant_BLACK = 10

test.rank_color.restype = ctypes.c_int
test.rank_color.argtypes = [enum_color]

test.best_color.restype = enum_color
test.best_color.argtypes = []

__anon_enum1 = ctypes.c_uint
enum_constant_READ = 0
enum_constant_WRITE = 1
enum_constant_APPEND = 2
